core-tests/jvm/src/test/scala/zio/IOSpec.scala

```diff
@@ -327,6 +328,20 @@ class IOSpec(implicit ee: org.specs2.concurrent.ExecutionEnv) extends TestRuntim
     }
   }
 
+  def testEventually = {
+    def effect(ref: Ref[Int]) =
+      ref.get.flatMap(n => if (n < 10) IO.fail("Ouch") else UIO.succeed(n))
+
+    val test = for {
+      ref   <- Ref.make(0)
+      fiber <- effect(ref).eventually.fork
+      _     <- ref.update(_ + 1).repeat(Schedule.recurs(9)).fork
+      n     <- fiber.join
```

Comment:
There is a possibility, that this test passes ocassionally without even using `eventually`. If forked fiber doesn't progress when line below executes completely.
I think it's possible to update `ref` (instead of `get`) in the `effect` and then you don't need undeterministic behaviour of `fork`. What do you think?

Reply:
Done. Sorry.